
// python3 -m grpc_tools.protoc -I. --python_out=. --grpc_python_out=. user.proto
// Run the above command in the folder where your user.proto is present. This will generate user_pb2.py and user_pb2_grpc.py files in the protos folder

/*

FLOW OF THIS PROOTO (CRUD):

-1- GET REQUEST: Requests user details by CNIC
-1- GET RESPONSE: Displays all the details of a certain user
-2- POST REQUEST: User Enters all of his details
-2- POST RESPONSE: Displays Acknowledgement or error message
-3- PUT REQUEST: User enters all the details again for updation, Password is (optional)
-3- PUT RESPONSE: Displays Acknowledgement or error message
-4- DELETE REQUEST: CNIC to identify the user to be deleted
-4- DELETE RESPONSE: Displays Acknowledgement or error message

*/

syntax = "proto3";

service UserService {
    rpc GetUserRPC (GetUserRequest) returns (GetUserResponse);
    rpc PostUserRPC (PostUserRequest) returns (PostUserResponse);
    rpc PutUserRPC (PutUserRequest) returns (PutUserResponse);
    rpc DeleteUserRPC (DeleteUserRequest) returns (DeleteUserResponse);
}

message GetUserRequest {
    int64 cnic = 1;
}

message GetUserResponse {
    int64 cnic = 1;
    string forename = 2;
    string surname = 3;
    string address = 4;
    string email = 5;
    string password = 6;
}

message PostUserRequest {
    int64 cnic = 1;
    string forename = 2;
    string surname = 3;
    string address = 4;
    string email = 5;
    string password = 6;
}

message PostUserResponse {
    string message = 1;
}

message PutUserRequest {
    int64 cnic = 1;
    string forename = 2;
    string surname = 3;
    string address = 4;
    string email = 5;
    string password = 6;
}

message PutUserResponse {
    string message = 1;
}

message DeleteUserRequest {
    int64 cnic = 1;
}

message DeleteUserResponse {
    string message = 1;
}
